stem/water_boundaries.py

```diff
@@ -0,0 +1,105 @@
+from typing import List, Union
+from dataclasses import dataclass, field
+from abc import ABC
+
+
+
+@dataclass
+class WaterBoundaryParameters(ABC):
+    """
+    Abstract base class for load water boundary parameters
+
+    Attributes:
+        - surfaces_assigment (List[str]): List of surfaces to which the water boundary is assigned.
+        - is_fixed (bool): True if the water boundary is fixed, False otherwise.
+        - gravity_direction (int): Direction of the gravity vector.
+        - out_of_plane_direction (int): Direction of the out of plane vector.
+
+
+    """
+    surfaces_assigment: List[str] = field(default_factory=lambda: [""])
+    is_fixed: bool = True
+    gravity_direction: int = 1
+    out_of_plane_direction: int = 2
+
+
+@dataclass
+class PhreaticMultiLineBoundary(WaterBoundaryParameters):
+    """
+    Class containing the load parameters for a phreatic line boundary condition
+
+    Attributes:
+        - x_coordinates (List[float]): X coordinates of the phreatic line [m].
+        - y_coordinates (List[float]): Y coordinates of the phreatic line [m].
+        - z_coordinates (List[float]): Z coordinates of the phreatic line [m].
+        - specific_weight (float): Specific weight of the water [kN/m3].
+
+
+    """
+    x_coordinates: List[float] = field(default_factory=lambda: [0.0])
```

Comment:
I think the parameters here should not be given as a default, as I think its important that the user actively thinks about what he/she wants

Reply:
except the z_coordinates, that is only required in 3D, so this can be checked in the post_init if z_coordinates are present in 3D cases

Reply:
The problem is that there are default values in the parent class. If a default value is defined in the parent class then everything else needs to have default values. This has to do with the ordering of the variables :/ I will make everything undefault but this is also not ideal